GitCommands/Git/Commands/GitCommandHelpers.cs

```diff
@@ -107,11 +108,11 @@ public static ArgumentString GetRefsCmd(bool tags, bool branches, bool noLocks,
                     ? (ArgumentString)"--no-optional-locks"
                     : default)
             {
-                { sortBy != GitRefsSortBy.Default, GetSortCriteria(tags, sortBy, sortOrder), string.Empty },
+                { sortBy != GitRefsSortBy.Default, GetSortCriteria(hasTags, sortBy, sortOrder), string.Empty },
                 format,
-                { branches, "refs/heads/", string.Empty },
-                { branches && tags, "refs/remotes/", string.Empty },
-                { tags, "refs/tags/", string.Empty },
+                { (getRef & GetRefsEnum.All) == 0 && (getRef & GetRefsEnum.Branches) != 0, "refs/heads/", string.Empty },
+                { (getRef & GetRefsEnum.All) == 0 && (getRef & GetRefsEnum.Remotes) != 0, "refs/remotes/", string.Empty },
+                { (getRef & GetRefsEnum.All) == 0 && (getRef & GetRefsEnum.Tags) != 0, "refs/tags/", string.Empty },
```

Comment:
We're duplicating this code, so it is worth considering extracting it in a local static method:
```cs
static bool HasFlag(GetRefsEnum requested, GetRefsEnum flag)
    => (requested & GetRefsEnum.All) == 0 && (requested & flag) != 0;
```

Reply:
changed